### ISSUE TITLE
＠fastmath does not handle +=, *=, etc.

### ISSUE THREAD

**Issue by simonster** [JuliaLang/julia#9778]
e.g.:

```
julia> f(a) = @fastmath a*a*a*a;

julia> g(a) = @fastmath a*=a*=a*=a;

julia> code_llvm(f, (Float64,))

define double @julia_f_199956(double) {
top:
  %1 = fmul fast double %0, %0, !dbg !301
  %2 = fmul fast double %1, %1, !dbg !301
  ret double %2, !dbg !301
}

julia> code_llvm(g, (Float64,))

define double @julia_g_199959(double) {
top:
  %1 = fmul double %0, %0, !dbg !310
  %2 = fmul double %1, %0, !dbg !310
  %3 = fmul double %2, %0, !dbg !310
  ret double %3, !dbg !310
}
```

I think this could probably be tweaked to work by lowering `a *= b` to `a = a * b` in the `@fastmath` macro, or we could possibly use whatever approach we pick for #8227.

**Comment by eschnett** [comment]
I thought that `+=` would be expanded before `@fastmath` gets its hands onto the expression... Apparently not. I'll add it. Thanks!

PS: Feel free to assign this issue to me.

**Comment by eschnett** [comment]
I have implemented this as part of #9759.
